allennlp/training/trainer.py

```diff
@@ -290,8 +297,130 @@ def on_end(
         **kwargs,
     ) -> None:
         if is_primary:
-            assert self._tensorboard is not None
-            self._tensorboard.close()
+            assert self._log_writer is not None
+            self._log_writer.close()
+
+
+@TrainerCallback.register("console-logger")
+class ConsoleLoggerCallback(TrainerCallback):
+    def __init__(
+        self,
+        serialization_dir: Optional[str] = None,
+        should_log_inputs: bool = False,
+    ) -> None:
+        super().__init__(serialization_dir=serialization_dir)
+        self._should_log_inputs = should_log_inputs
+
+    def on_batch(
+        self,
+        trainer: "GradientDescentTrainer",
+        batch_inputs: List[List[TensorDict]],
+        batch_outputs: List[Dict[str, Any]],
+        batch_metrics: Dict[str, Any],
+        epoch: int,
+        batch_number: int,
+        is_training: bool,
+        is_primary: bool = True,
+        batch_grad_norm: Optional[float] = None,
+        **kwargs,
+    ) -> None:
+        # In the distributed case we need to call this from every worker, since every
+        # worker reports its own memory usage.
+        cpu_memory_usage = common_util.peak_cpu_memory()
+        gpu_memory_usage = common_util.peak_gpu_memory()
```

Comment:
We probably don't want to call this on every batch, especially if we're only logging it on the first batch

Reply:
See comment below as well.